bundles/core/src/main/java/com/adobe/cq/commerce/demandware/preview/PreviewServiceConfigImpl.java

```diff
@@ -0,0 +1,155 @@
+    
+    @Property(label = "Enable storefront protection", boolValue = false)
+    private static final String STORFRONT_PROTECTION_ENABLED = "storefront.protected";
+    
+    @Property(label = "Protected storefront user")
+    private static final String STORFRONT_PROTECTION_USER = "storefront.user";
+    
+    @Property(label = "Protected storefront password")
+    private static final String STORFRONT_PROTECTION_PASSWORD = "storefront.password";
+    
+    @Property(label = "Instance id", description = "Preview Service instance id that corresponds to Replication Agent config")
+    private static final String INSTANCE_ID = "instance.id";
+    
+    private String previewPageEndPoint;
+    private String previewSearchEndPoint;
+    private String previewTemplate;
+    private String previewDefaultSite;
+    private String previewCacheEnabled;
+    private String previewCacheTime;
+    private String storfrontProtectionEnabled;
+    private String storfrontProtectionUser;
+    private String StorfrontProtectionPassword;
+    private String instanceId;
+    
+    private PreviewCache cache;
+    
+    public String getInstanceId() {
+        return instanceId;
+    }
+    
+    public String getPreviewPageEndPoint() {
+        return previewPageEndPoint;
+    }
+    
+    public String getPreviewSearchEndPoint() {
+        return previewSearchEndPoint;
+    }
+    
+    public String getPreviewTemplate() {
+        return previewTemplate;
+    }
+    
+    public String getPreviewDefaultSite() {
+        return previewDefaultSite;
+    }
+    
+    public String getPreviewCacheEnabled() {
+        return previewCacheEnabled;
+    }
+    
+    public String getPreviewCacheTime() {
+        return previewCacheTime;
+    }
+    
+    public String getStorfrontProtectionEnabled() {
+        return storfrontProtectionEnabled;
+    }
+    
+    public String getStorfrontProtectionUser() {
+        return storfrontProtectionUser;
+    }
+    
+    public String getStorfrontProtectionPassword() {
+        return StorfrontProtectionPassword;
+    }
+    
+    public PreviewCache getCache() {
+        return cache;
+    }
+    
+    @Activate
+    protected void activate(Map<String, Object> configuration) {
+        previewPageEndPoint = StringUtils.prependIfMissing(PropertiesUtil.toString(configuration.get
+                (PREVIEW_PAGE_ENDPOINT), null), "/", "/");
+        previewSearchEndPoint = StringUtils.prependIfMissing(
+                PropertiesUtil.toString(configuration.get(PREVIEW_SEARCH_ENDPOINT), null), "/", "/");
+        previewTemplate = PropertiesUtil.toString(configuration.get(PREVIEW_TEMPLATE), null);
+        previewDefaultSite = PropertiesUtil.toString(configuration.get(PREVIEW_DEFAULT_SITE), null);
+        previewCacheTime = PropertiesUtil.toString(configuration.get(PREVIEW_CACHE_TIME), "60");
+        storfrontProtectionEnabled = PropertiesUtil.toString(configuration.get(PREVIEW_CACHE_TIME), null);
```

Comment:
storfrontProtectionEnabled can be fetched as a boolean (see source code before), then you don't need to convert it later